adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,479 @@
+
+_Returns_: A [code]#work_item# representing the calling work-item within group
+[code]#g#.
+
+'''
+
+.[apidef]#leader_of#
+[source,role=synopsis,id=api:common-group-leader_of]
+----
+template <typename Group>
+bool leader_of(Group g) noexcept;
+----
+
+_Returns_: [code]#true# if the calling work-item is the leader of group
+[code]#g#, and [code]#false# otherwise.
+
+_Remarks_: [code]#leader_of# returns [code]#true# for only one work-item in a
+group.
+The leader of the group is determined during construction of the group, and is
+invariant for the lifetime of the group.
+The leader of the group is guaranteed to be the work-item with index 0 within
+the group.
+
+[[sec:khr-group-interface-work_group]]
+== [code]#work_group# class
+
+The [code]#work_group# class template encapsulates all functionality required to
+represent a particular <<work-group>> within a kernel.
+It is not user-constructible.
+
+The SYCL [code]#work_group# class template provides common by-value semantics
+(see <<sec:byval-semantics>>) and the common group interface (see
+<<sec:khr-group-interface-common>>).
+
+[source,role=synopsis]
+----
+namespace sycl {
+namespace khr {
+
+template <int Dimensions = 1>
+class work_group {
+
+ public:
+  using id_type = id<Dimensions>;
+  using linear_id_type = size_t;
+  using range_type = range<Dimensions>;
+#if defined(__cpp_lib_mdspan)
+  using extents_type = std::dextents<size_t, Dimensions>;
+#endif
+  using size_type = size_t;
+  static constexpr int dimensions = Dimensions;
+  static constexpr memory_scope fence_scope = memory_scope::work_group;
+
+  work_group(group<Dimensions> g) noexcept;
+
+  operator group<Dimensions>() const noexcept;
+
+  /* -- common by-value interface members -- */
+
+  /* -- common group interface members -- */
+
+};
+
+} // namespace khr
+} // namespace sycl
+----
+
+.[apidef]#work_group constructor#
+[source,role=synopsis,id=api:khr-group-interface-work-group-constructor]
+----
+work_group(group<Dimensions> g) noexcept;
+----
+
+_Effects_: Constructs a [code]#work_group# representing the same collection of
+work-items as [code]#g#.
+
+'''
+
+.[apidef]#work_group conversion operator#
+[source,role=synopsis,id=api:khr-group-interface-work-group-conversion-operator]
+----
+operator group<Dimensions>() const noexcept;
+----
+
+_Returns_: A [code]#group# representing the same collection of work-items as
+this [code]#work_group#.
+
+'''
+
+[[sec:khr-group-interface-sub_group]]
+== [code]#sub_group# class
+
+The [code]#sub_group# class template encapsulates all functionality required to
+represent a particular <<sub-group>> within a kernel.
+It is not user-constructible.
+
+The SYCL [code]#sub_group# class template provides common by-value semantics
+(see <<sec:byval-semantics>>) and the common group interface (see
+<<sec:khr-group-interface-common>>).
+
+[source,role=synopsis]
+----
+namespace sycl {
+namespace khr {
+
+class sub_group {
```

Comment:
What is your naming plan if this KHR is eventually adopted into the core spec?  We already have a type named `sub_group`.

Reply:
The plan is to declare equivalents of all the members of `sycl::khr::sub_group` inside of `sycl::sub_group`, and deprecate the existing `sycl::sub_group` interface.

I appreciate this is an awkward way to go about it, but this KHR cannot add the necessary functions directly to `sycl::sub_group`.  Our naming convention dictates that any such functions be called things like `sycl::sub_group::khr_id()` -- in addition to being different from what we're actually trying to propose, it would be different to the interface exposed by `sycl::khr::work_group` and `sycl::khr::work_item`.